src/main/java/amazin/controller/BookController.java

```diff
@@ -93,12 +96,24 @@ public String deleteBook(@PathVariable("id") Long id, RedirectAttributes attribu
         Optional<Book> deletedBook = bookService.delete(id);
 
         if (deletedBook.isPresent()) {
-            attributes.addAttribute(PARAMETER_BOOK_ID, deletedBook.get().getId());
+            Book book = deletedBook.get();
+            deleteFile(book.getPicture_url());
+            attributes.addAttribute(PARAMETER_BOOK_ID, book.getId());
         }
 
         return createRedirectViewPath(REQUEST_MAPPING_BOOK);
     }
 
+    @PostMapping("/storage/uploadFile")
+    public String uploadFile(@RequestPart(value = "file") MultipartFile file) {
```

Comment:
Are we not restricting file size? or file type?

Reply:
File type is restricted in the form.
Plus the maximum an attacker can do in terms of uploading a big image is increase our amazon bill :p